lib/generators/boring/devise/jwt/install/install_generator.rb

```diff
@@ -0,0 +1,224 @@
+
+        def verify_presence_of_devise_gem
+          gem_file_content_array = File.readlines("Gemfile")
+          devise_is_installed = gem_file_content_array.any? { |line| line.include?('devise') }
+
+          return if devise_is_installed
+
+          say "We couldn't find devise gem. Please configure devise gem and run the generator again!", :red
+
+          abort
+        end
+
+        def verify_presence_of_devise_initializer
+          return if File.exist?("config/initializers/devise.rb")
+
+          say "We couldn't find devise initializer. Please configure devise gem correctly and run the generator again!", :red
+
+          abort
+        end
+
+        def verify_presence_of_devise_model
+          return if File.exist?("app/models/#{options[:model_name].underscore}.rb")
+
+          say "We couldn't find the #{options[:model_name]} model. Maybe there is a typo? Please provide the correct model name and run the generator again.", :red
+
+          abort
+        end
+
+        def add_devise_jwt_gem
+          say "Adding devise-jwt gem", :green
+          gem "devise-jwt"
+        end
+
+        def add_devise_jwt_config_to_devise_initializer
+          say "Adding devise-jwt configurations to a file `config/initializers/devise.rb`", :green
+
+          jwt_config = <<~RUBY
+            config.jwt do |jwt|
+              jwt.secret = #{devise_jwt_secret_key}
+              jwt.dispatch_requests = [
+                ['POST', %r{^/sign_in$}]
+              ]
+              jwt.revocation_requests = [
+                ['DELETE', %r{^/sign_out$}]
+              ]
+              jwt.expiration_time = #{options[:expiration_time_in_days]}.day.to_i
+            end
+          RUBY
+
+          inject_into_file "config/initializers/devise.rb",
+                            optimize_indentation(jwt_config, 2),
+                            before: /^end\s*\Z/m
+
+          say "❗️❗️\nValue for jwt.secret will be used from `#{devise_jwt_secret_key}`. You can change this values if they don't match with your app.\n",
+              :yellow
+        end
+
+        def configure_revocation_strategies
+          say "Configuring #{options[:revocation_strategy]} revocation strategy",
+              :green
+
+          case options[:revocation_strategy]
+          when "JTIMatcher"
+            configure_jti_matcher_strategy
+          when "Denylist"
+            configure_denylist_strategy
+          when "Allowlist"
+            configure_allowlist_strategy
+          end
+        end
+
+        private
+
+        def devise_jwt_secret_key
+          if options[:use_env_variable]
+            "ENV['DEVISE_JWT_SECRET_KEY']"
+          else
+            "Rails.application.credentials.devise_jwt_secret_key"
+          end
+        end
+
+        def configure_jti_matcher_strategy
+          model_name = options[:model_name].underscore
+          Bundler.with_unbundled_env do
```

Comment:
can you use migration template here instead? I think that would be easier in long run?

Reply:
I find it challenging to determine the next migration number for the migration file name. Do you have any suggestions for a better approach? I am considering using the current timestamp.

@abhaynikam

Reply:
I find this function on rails source code
```
        def self.next_migration_number(dirname)
          next_migration_number = current_migration_number(dirname) + 1
          ActiveRecord::Migration.next_migration_number(next_migration_number)
        end
```
can I use this and use migration template_method by including Rails::Generators::Migration ?

Reply:
here is the documentation of migration_template

```
Creates a migration template at the given destination. The difference to the default template method is that the migration version is appended to the destination file name.

The migration version, migration file name, migration class name are available as instance variables in the template to be rendered.
```